comment/models/comments.py

```diff
@@ -49,10 +51,21 @@ def _set_unique_urlhash(self):
             while self.__class__.objects.filter(urlhash=self.urlhash).exists():
                 self.urlhash = self.__class__.objects.generate_urlhash()
 
+    def _set_email(self):
+        if self.user:
+            self.email = self.user.email
+
     def save(self, *args, **kwargs):
         self._set_unique_urlhash()
+        self._set_email()
         super(Comment, self).save(*args, **kwargs)
 
+    def get_absolute_url(self):
+        page_url = self.content_object.get_absolute_url()
+        if not page_url.endswith('/'):
+            page_url += '/'
+        return page_url + '#' + self.urlhash
```

Comment:
This is also not correct. There needs to be some consideration for pagination but since the comments per page is retrieved from the template, I currently, don't understand how this can be fixed here.

Reply:
Won't `self.urlhash` take you to the comment regarless the pagination? If the answer NO, then a dirty solution is that we may be able to know in which page the comment is based on posted time.

Reply:
> Won't self.urlhash take you to the comment regarless the pagination

No, the comment won't be present on the page due to pagination.

> a dirty solution is that we may be able to know in which page the comment is based on posted time

As far as I remember, comments per page is obtained from the template. Without it, how can we know the page number?

Reply:
Moving pagination vars to settings?

Reply:
yes, this can be one of the solutions.

Reply:
You are right. But this won't significantly affect the app.